terragrunt/env/staging/api/terragrunt.hcl

```diff
@@ -2,6 +2,39 @@ terraform {
   source = "../../../aws//api"
 }
 
+dependencies {
+  paths = ["../network", "../ecr"]
+}
+
+dependency "network" {
+  config_path = "../network"
+
+  mock_outputs_allowed_terraform_commands = ["init", "fmt", "validate", "plan", "show"]
+  mock_outputs_merge_with_state           = true
+  mock_outputs = {
+    api_security_group_id = ""
+    private_subnet_ids = [""]
+  }
+}
+
+dependency "ecr" {
+  config_path = "../ecr"
+
+  mock_outputs_allowed_terraform_commands = ["init", "fmt", "validate", "plan", "show"]
+  mock_outputs_merge_with_state           = true
+  mock_outputs = {
+    ecr_repository_arn = ""
+    ecr_repository_url = ""
+  }
+}
+
+inputs = {
+  api_security_group_id = dependency.network.outputs.api_security_group_id
```

Comment:
This is one of the errors you are hitting - in your `inputs.tf` you've got the variable defined as `api_id`, but here you're trying to set a non-existent variable.  

Luckily, this is a perfect name for that variable so all you need to do is update what you've called it in `inputs.tf`.

Reply:
Done!